singularity-eos/eos/eos.hpp

```diff
@@ -125,10 +125,19 @@ class Gruneisen : public EosBase<Gruneisen> {
  public:
   Gruneisen() = default;
   PORTABLE_INLINE_FUNCTION
+  Gruneisen(const Real C0, const Real s1, const Real s2, const Real s3, const Real G0,
+            const Real b, const Real rho0, const Real T0, const Real P0, const Real Cv,
+            const Real rho_max)
+      : _C0(C0), _s1(s1), _s2(s2), _s3(s3), _G0(G0), _b(b), _rho0(rho0), _T0(T0), _P0(P0),
+        _Cv(Cv), _rho_max(rho_max) {}
+  // Constructor when rho_max isn't specified automatically determines _rho_max
+  PORTABLE_INLINE_FUNCTION
   Gruneisen(const Real C0, const Real s1, const Real s2, const Real s3, const Real G0,
             const Real b, const Real rho0, const Real T0, const Real P0, const Real Cv)
       : _C0(C0), _s1(s1), _s2(s2), _s3(s3), _G0(G0), _b(b), _rho0(rho0), _T0(T0), _P0(P0),
-        _Cv(Cv) {}
+        _Cv(Cv), _rho_max(0.99 * GetRhoMax(s1, s2, s3, rho0)) {}
```

Comment:
I assume the 0.99 is a safety factor?

Reply:
Might be good to define a static class variable `static constexpr Real RHOMAX_SAFETY = 0.99` or something... and maybe a comment indicating this is consistent with Pagosa.